src/field_theory/primitive_element.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2020 Thomas Browning and Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning and Patrick Lutz
+-/
+
+import field_theory.adjoin
+import field_theory.separable
+
+/-!
+# Primitive Element Theorem
+
+In this file we prove the primitive element theorem.
+
+## Main results
+
+- `primitive_element`: a finite separable extension has a primitive element.
+
+-/
+
+noncomputable theory
+open_locale classical
+
+/- Proof of the primitive element theorem. -/
+
+open finite_dimensional
+
+namespace field
+
+section primitive_element_finite
+variables (F : Type*) [field F] {E : Type*} [field E] [algebra F E]
+
+/-! ### Primitive element theorem for finite fields -/
+
+/-- Primitive element theorem assuming E is finite. -/
+lemma primitive_element_fin_aux [fintype E] : ∃ α : E, F⟮α⟯ = ⊤ :=
```

Comment:
I'm not sure if I like this name. The theorem immediately below it also assumes that something is finite (`F` rather than `E`). Would you be okay with `primitive_element_of_fintype_top`?